### ISSUE TITLE
Performance issue with the built-in ChangeRule

### ISSUE THREAD

**Issue by kyatapi** [Yelp/elastalert#2340]
The code in the 'compare' function of ChangeRule class is causing significant performance loss, especially when the set of 'query_key' is huge.

    def compare(self, event):
        key = hashable(lookup_es_key(event, self.rules['query_key']))
        values = []
        elastalert_logger.debug(" Previous Values of compare keys  " + str(self.occurrences))
        for val in self.rules['compound_compare_key']:
            lookup_value = lookup_es_key(event, val)
            values.append(lookup_value)
        elastalert_logger.debug(" Current Values of compare keys   " + str(values))

Please refer to the 4th line of the above code segment, converting the whole **self.occurrences** to string is unnecessary, as what's important here is merely the previous values of current **key** instead of the whole. Also, the conversion will be executed no matter what the logging level is, as it is done before passing the composed string to the **elastalert_logger.debug** function.

**Comment by Qmando** [comment]
Good point. I hadn't really considered the performance of this piece. I'll put it behind a check for the logging level.
